### ISSUE TITLE
[Bug]: Orphan hosts not getting pruned 

### ISSUE THREAD

**Issue by CreeperFace00** [netdata/netdata#16717]
### Bug description

Recently I have noticed a lot of hosts that I have decommissioned many months ago have reappeared in the list of replicated nodes on my parent.
This seems to be affecting all versions of the dashboard.

In `netdata.conf` on my parent node I have configured `cleanup orphan hosts after secs = 21600` and `delete orphan hosts files = yes`

### Expected behavior

Child nodes that are no longer streaming data should be removed from the DB.

### Steps to reproduce

1. Stream metrics from child to parent node
2. Disable streaming on child node
3. Wait the duration specified in `cleanup orphan hosts after secs`
4. Observe dead nodes not being removed from the dashboard.

### Installation method

from git

### System info

```shell
max@netdata  ~  uname -a; grep -HvE "^#|URL" /etc/*release
Linux netdata 5.15.0-91-generic #101-Ubuntu SMP Tue Nov 14 13:30:08 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=22.04
/etc/lsb-release:DISTRIB_CODENAME=jammy
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 22.04.3 LTS"
/etc/os-release:PRETTY_NAME="Ubuntu 22.04.3 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION_ID="22.04"
/etc/os-release:VERSION="22.04.3 LTS (Jammy Jellyfish)"
/etc/os-release:VERSION_CODENAME=jammy
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:UBUNTU_CODENAME=jammy
```

### Netdata build info

```shell
netdata -W buildinfo
Packaging:
    Netdata Version ____________________________________________ : v1.44.0-122-nightly
    Installation Type __________________________________________ : custom
    Package Architecture _______________________________________ : unknown
    Package Distro _____________________________________________ : unknown
    Configure Options __________________________________________ : dummy-configure-command
Default Directories:
    User Configurations ________________________________________ : /etc/netdata
    Stock Configurations _______________________________________ : /usr/lib/netdata/conf.d
    Ephemeral Databases (metrics data, metadata) _______________ : /var/cache/netdata
    Permanent Databases ________________________________________ : /var/lib/netdata
    Plugins ____________________________________________________ : /usr/libexec/netdata/plugins.d
    Static Web Files ___________________________________________ : /usr/share/netdata/web
    Log Files __________________________________________________ : /var/log/netdata
    Lock Files _________________________________________________ : /var/lib/netdata/lock
    Home _______________________________________________________ : /var/lib/netdata
Operating System:
    Kernel _____________________________________________________ : Linux
    Kernel Version _____________________________________________ : 5.15.0-91-generic
    Operating System ___________________________________________ : Ubuntu
    Operating System ID ________________________________________ : ubuntu
    Operating System ID Like ___________________________________ : debian
    Operating System Version ___________________________________ : 22.04.3 LTS (Jammy Jellyfish)
    Operating System Version ID ________________________________ : none
    Detection __________________________________________________ : /etc/os-release
Hardware:
    CPU Cores __________________________________________________ : 11
    CPU Frequency ______________________________________________ : 2200000000
    RAM Bytes __________________________________________________ : 16723726336
    Disk Capacity ______________________________________________ : 107374182400
    CPU Architecture ___________________________________________ : x86_64
    Virtualization Technology __________________________________ : kvm
    Virtualization Detection ___________________________________ : systemd-detect-virt
Container:
    Container __________________________________________________ : none
    Container Detection ________________________________________ : systemd-detect-virt
    Container Orchestrator _____________________________________ : none
    Container Operating System _________________________________ : none
    Container Operating System ID ______________________________ : none
    Container Operating System ID Like _________________________ : none
    Container Operating System Version _________________________ : none
    Container Operating System Version ID ______________________ : none
    Container Operating System Detection _______________________ : none
Features:
    Built For __________________________________________________ : Linux
    Netdata Cloud ______________________________________________ : YES
    Health (trigger alerts and send notifications) _____________ : YES
    Streaming (stream metrics to parent Netdata servers) _______ : YES
    Back-filling (of higher database tiers) ____________________ : YES
    Replication (fill the gaps of parent Netdata servers) ______ : YES
    Streaming and Replication Compression ______________________ : YES (zstd lz4 gzip)
    Contexts (index all active and archived metrics) ___________ : YES
    Tiering (multiple dbs with different metrics resolution) ___ : YES (5)
    Machine Learning ___________________________________________ : YES
Database Engines:
    dbengine ___________________________________________________ : YES
    alloc ______________________________________________________ : YES
    ram ________________________________________________________ : YES
    map ________________________________________________________ : YES
    save _______________________________________________________ : YES
    none _______________________________________________________ : YES
Connectivity Capabilities:
    ACLK (Agent-Cloud Link: MQTT over WebSockets over TLS) _____ : YES
    static (Netdata internal web server) _______________________ : YES
    h2o (web server) ___________________________________________ : YES
    WebRTC (experimental) ______________________________________ : NO
    Native HTTPS (TLS Support) _________________________________ : YES
    TLS Host Verification ______________________________________ : YES
Libraries:
    LZ4 (extremely fast lossless compression algorithm) ________ : YES
    ZSTD (fast, lossless compression algorithm) ________________ : YES
    zlib (lossless data-compression library) ___________________ : YES
    protobuf (platform-neutral data serialization protocol) ____ : YES (bundled)
    OpenSSL (cryptography) _____________________________________ : YES
    libdatachannel (stand-alone WebRTC data channels) __________ : NO
    JSON-C (lightweight JSON manipulation) _____________________ : YES
    libcap (Linux capabilities system operations) ______________ : YES
    libcrypto (cryptographic functions) ________________________ : YES
Plugins:
    apps (monitor processes) ___________________________________ : YES
    cgroups (monitor containers and VMs) _______________________ : YES
    cgroup-network (associate interfaces to CGROUPS) ___________ : YES
    proc (monitor Linux systems) _______________________________ : YES
    tc (monitor Linux network QoS) _____________________________ : YES
    diskspace (monitor Linux mount points) _____________________ : YES
    freebsd (monitor FreeBSD systems) __________________________ : NO
    macos (monitor MacOS systems) ______________________________ : NO
    statsd (collect custom application metrics) ________________ : YES
    timex (check system clock synchronization) _________________ : YES
    idlejitter (check system latency and jitter) _______________ : YES
    bash (support shell data collection jobs - charts.d) _______ : YES
    debugfs (kernel debugging metrics) _________________________ : YES
    cups (monitor printers and print jobs) _____________________ : NO
    ebpf (monitor system calls) ________________________________ : YES
    freeipmi (monitor enterprise server H/W) ___________________ : NO
    nfacct (gather netfilter accounting) _______________________ : NO
    perf (collect kernel performance events) ___________________ : YES
    slabinfo (monitor kernel object caching) ___________________ : YES
    Xen ________________________________________________________ : NO
    Xen VBD Error Tracking _____________________________________ : NO
    Logs Management ____________________________________________ : YES
Exporters:
    AWS Kinesis ________________________________________________ : NO
    GCP PubSub _________________________________________________ : NO
    MongoDB ____________________________________________________ : NO
    Prometheus (OpenMetrics) Exporter __________________________ : YES
    Prometheus Remote Write ____________________________________ : NO
    Graphite ___________________________________________________ : YES
    Graphite HTTP / HTTPS ______________________________________ : YES
    JSON _______________________________________________________ : YES
    JSON HTTP / HTTPS __________________________________________ : YES
    OpenTSDB ___________________________________________________ : YES
    OpenTSDB HTTP / HTTPS ______________________________________ : YES
    All Metrics API ____________________________________________ : YES
    Shell (use metrics in shell scripts) _______________________ : YES
Debug/Developer Features:
    Trace All Netdata Allocations (with charts) ________________ : NO
    Developer Mode (more runtime checks, slower) _______________ : YES
```

### Additional info

I currently have a debug build installed as I am trying to track down a (probably) unrelated crashing issue. 

I am also streaming metrics from an access point. It seems the unique host ID does not persist through a device reboot causing my replication list to look like this:
 
![image](https://github.com/netdata/netdata/assets/16279973/7d7cf17b-e88a-450b-ba5e-10490f71ee48)

This duplicate host issue does not seem to impact the new dashboard.

**Comment by ilyam8** [comment]
I think that is not a bug but by design. Netdata will remove children hosts metadata **only if you mark them ephemeral** (default is no, supported only in nightly versions), this can be done in:

- `netdata.conf`

```
[global]
  is ephemeral node = yes
```

- `stream.conf` (all nodes that connect using a specific API key)

```
[API_KEY]
  is ephemeral node = yes
```

Cleanup timing is configured in parent's `netdata.conf`

```
[db]
  cleanup ephemeral hosts after secs = 86400
```

@stelfrag correct me if I'm wrong.

---

> It seems the unique host ID does not persist through a device reboot

This ID is `/var/lib/netdata/registry/netdata.public.unique.id`. It's not a Netdata error, if it doesn't persist, you need to check why it's not happening.

**Comment by CreeperFace00** [comment]
Hi ilyam8,

I've added `is ephemeral node = yes` to the access point that is duplicating itself, I'll pursue some solutions to it's weird behavior. I'm aware that issue is separate from Netdata itself, but I believe the issue of this device's previous entries not being removed to be a symptom of the cleanup issue I am seeing. 


> Cleanup timing is configured in parent's netdata.conf

I do have cleanup configured in netdata.conf, the issue is it appears to be having no effect.
Up until sometime a couple months ago the cleanup function did a good job of removing dead hosts, but currently I am seeing dead hosts that are many months old persisting despite having this configured in netdata.conf.

This is what I have in netdata.conf:
```
cleanup orphan hosts after secs = 3600
delete orphan hosts files = yes
cleanup ephemeral hosts after secs = 1800
```

**Comment by ilyam8** [comment]
>  that is duplicating itself

You need to find out why `/var/lib/netdata/registry/netdata.public.unique.id` on the child instance is different after reboot. We can't help with that.

**Comment by CreeperFace00** [comment]
Yep, I'm aware that is unrelated, I think there is a miscommunication here. Let's ignore the duplicate device issue completely, it's an external problem, completely irrelevant to the actual issue I am seeing. 
 

The actual issue is that I have many dead hosts that are not being cleaned after they have been offline for several months.
![image](https://github.com/netdata/netdata/assets/16279973/19916b79-b1c7-41c5-80aa-a896d36e8911)
These particular hosts were removed from service back in September, but still persist despite having cleanup configured.

**Comment by CreeperFace00** [comment]
In the end I completely erased my dbengine database and am starting from scratch. 
I'm almost positive all of my issues have been caused by some kind of database corruption, as my Netdata parent required restarting 10+ times per day. I searched through the issues on github to see if anyone else had the problems I was having, luckily it looks like it was just something with my weird setup.

Closing this issue, unless anybody would like to look through my problematic database files.
